### PR TITLE
Add GA click events to navigation.js

### PR DESCRIPTION
**Pull request by caldav**
## Done

* GA event framework to capture:
  * Nav and footer usage
  * Link clicks in content (charm cards, listing navigation, etc.)

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://localhost:8045
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Open GA on the charmhub.io or combo tracker property -> Real time view -> Refine to yourself using location filtering if possible -> Events. Otherwise, you can use the GA debugger [(Chrome/Chromium only)](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) and look at console output.
- Ensure the charmhub.io cookie policy is accepted at least at the "performance" level.
- Use the site: ensure clicking on charm card is sending a card event, click links, use the main navigation, navigate between tabs of a listing.

## Screenshots

Screenshot from 2020-11-18 13-02-32

Screenshot from 2020-11-18 13-03-24

### CODE REVIEW

File: static/js/src/base/navigation.js [hunk 1]
```diff
@@ -88,3 +88,66 @@ document.addEventListener("click", function (event) {
 });
 
 export { toggleSubnav };
+
+var origin = window.location.href;
```
Comment (tbille) [inline static/js/src/base/navigation.js:92]: @steverydz do you think it could go on it own file?

Comment (steverydz) [reply static/js/src/base/navigation.js:92]: @tbille yes, I think that would make sense. It could be `static/js/src/publisher/analytics.js`

Comment (caldav) [reply static/js/src/base/navigation.js:92]: As an FYI, ad blockers are actively blocking JS files named "analytics.js", "GA.js", etc.

Comment (caldav) [reply static/js/src/base/navigation.js:92]: What about `/publisher/navigation-events.js`?

Comment (steverydz) [reply static/js/src/base/navigation.js:92]: `navigation-events.js` sounds good. The path should be `static/js/src/public/`, my mistake.

**Comment by caldav** [comment]
Thanks for the update @steverydz , ran QA again, still working well

**Review by tbille — APPROVED** [review]
Let's merge it. Thanks @steverydz and @caldav

**Pull request merged**